### PR TITLE
Add main.py and update example_config.yaml

### LINKED ISSUE #11: Update env.yml and examples_config.yaml
Update [env.yml](https://github.com/cmlab-mira/template/blob/master/env.yml)  (delete the prefix section) and [examples_config.yaml](https://github.com/cmlab-mira/template/blob/master/src/configs/example_config.yaml) after merging all PRs.

### PR DESCRIPTION
**Pull request by peter850706**
Resolve #11.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    # Start training.
+    logging.info('Start training.')
+    trainer.train()
+    logging.info('End training.')
+
+
+def _parse_args():
+    parser = argparse.ArgumentParser(description="The script to train.")
+    parser.add_argument('--config_file', '-c', type=Path, help='The path of the config file.')
```
Comment (vic85821) [inline src/main.py]: Need to set required = True?

File: src/main.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    torch.cuda.manual_seed_all(random.getstate()[1][1])
+    torch.backends.cudnn.deterministic = True
+    torch.backends.cudnn.benchmark = False
+
+    logging.info('Create the device.')
+    if 'cuda' in config.trainer.device and not torch.cuda.is_available():
+        raise ValueError("The cuda is not available. Please set the device in the trainer section to 'cpu'.")
+    device = torch.device(config.trainer.device)
+
+    logging.info('Create the train and validation datasets.')
```
Comment (vic85821) [inline src/main.py]: train -> training?

File: src/main.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    train_dataset = _get_instance(src.data.datasets, config.dataset, 'train')
+    valid_dataset = _get_instance(src.data.datasets, config.dataset, 'valid')
+
+    logging.info('Create the train and validation dataloaders.')
```
Comment (vic85821) [inline src/main.py]: As previous

File: src/main.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    cls = getattr(src.data.datasets, config.dataset.name)
+    config.dataloader.kwargs.update(collate_fn=getattr(cls, 'collate_fn', None))
+    train_dataloader = _get_instance(src.data.dataloader, config.dataloader, train_dataset)
+    valid_dataloader = _get_instance(src.data.dataloader, config.dataloader, valid_dataset)
```
Comment (vic85821) [inline src/main.py:43]: The batch sizes of the training dataloader and the validation dataloader may be different?

File: src/main.py (hunk 1 above)
Comment (peter850706) [reply src/main.py]: Yes, or `parser.add_argument('config_file', type=Path, help='The path of the config file.')`?

File: src/main.py (hunk 2 above)
Comment (peter850706) [reply src/main.py]: Ok.

File: src/main.py (hunk 3 above)
Comment (peter850706) [reply src/main.py]: Ok.

File: src/main.py (hunk 1 above)
Comment (vic85821) [reply src/main.py]: Both can work.

File: src/main.py (hunk 4 above)
Comment (peter850706) [reply src/main.py:43]: Yes, so we need kwargs `train_batch_size` and `valid_batch_size` in the dataloader section of the config.

Comment (vic85821) [reply src/main.py:43]: OK

File: src/main.py (hunk 1 above)
Comment (peter850706) [reply src/main.py]: Prefer the last one, have updated.

File: src/main.py (hunk 4 above)
Comment (peter850706) [reply src/main.py:43]: Have updated.

**Pull request merged**